java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -91,4 +89,119 @@ protected static ConnectionManager buildConnectionManager(ChannelHandler channel
     protected static CommandManager buildCommandManager(ChannelHandler channelHandler) {
         return new CommandManager(channelHandler);
     }
+
+    /**
+     * Extracts the response from the Protobuf response and either throws an exception or returns the
+     * appropriate response as an <code>Object</code>.
+     *
+     * @param response Redis protobuf message
+     * @return Response <code>Object</code>
+     */
+    protected Object handleObjectResponse(Response response) {
+        // convert protobuf response into Object
+        return new BaseCommandResponseResolver(RedisValueResolver::valueFromPointer).apply(response);
+    }
+
+    /**
+     * Checks that the Response contains the simple-string Ok.
+     *
+     * @return An Ok response
+     * @throws RedisException if there's a type mismatch
+     */
+    protected Ok handleOkResponse(Response response) {
+        Object value = handleObjectResponse(response);
+        if (value == Ok.INSTANCE) {
+            return Ok.INSTANCE;
+        }
+        String className = (value == null) ? "null" : value.getClass().getSimpleName();
+        throw new RedisException(
```

Comment:
If this was non-null, should this log the actual value's toString() rather than just the class name?

Reply:
Better not to expose user data to our logging?